### PR TITLE
Help Command // improved 'help' and error messages

### PR DESCRIPTION
**Pull request by philip1304**
Help Command // improved 'help' and error messages

Notes:
- Displays better example usage for every command.
- Added 'edit' command.

### CODE REVIEW

File: src/main/java/seedu/pill/command/HelpCommand.java [hunk 1]
```diff
@@ -41,44 +49,125 @@ private void showGeneralHelp() {
...
+            showListHelp();
             break;
         case "quit":
-            System.out.println("quit: Exits the program.");
-            System.out.println("Usage: quit");
+            showQuitHelp();
             break;
         default:
             System.out.println("Unknown command: " + command);
-            System.out.println("Type 'help' for a list of available commands.");
```
Comment (yijiano) [inline src/main/java/seedu/pill/command/HelpCommand.java:81]: I appreciate the decision to abstract out the print implementation, makes out code more readable.

File: src/main/java/seedu/pill/command/HelpCommand.java [hunk 2]
```diff
@@ -41,44 +49,125 @@ private void showGeneralHelp() {
...
+    private void showHelpHelp() {
+        System.out.println("help: Shows help information about available commands.");
+        System.out.println("Usage: help [command] [-v]");
+        System.out.println("  [command] - Optional. Specify a command to get detailed help.");
+        System.out.println("  [-v]      - Optional. Show verbose output with examples.");
+        if (verbose) {
+            System.out.println("\nExamples:");
+            System.out.println("  help");
+            System.out.println("  help add");
+            System.out.println("  help add -v");
```
Comment (yijiano) [inline src/main/java/seedu/pill/command/HelpCommand.java:102]: Thanks for including the verbose parameter. Small nit would be to move the description of the command parameters to only show when verbose is true, but I don't think it matters that much.

**Review by yijiano — APPROVED** [review]
LGTM. Will add the verbose option to the parser soon.

File: src/main/java/seedu/pill/command/HelpCommand.java (hunk 2 above)
Comment (philip1304) [reply src/main/java/seedu/pill/command/HelpCommand.java:102]: I have some ideas for that actually, I'll make some changes in my next PR.

**Pull request merged**